include/qt_atomics.h

```diff
@@ -149,18 +149,20 @@ extern pthread_mutexattr_t _fastlock_attr;
 # define QTHREAD_TRYLOCK_INIT(x)     { (x).u = 0; }
 # define QTHREAD_TRYLOCK_INIT_PTR(x) { (x)->u = 0; }
 # define QTHREAD_TRYLOCK_LOCK(x)     { uint32_t val = qthread_incr(&(x)->s.users, 1); \
-                                       while (val != (x)->s.ticket) SPINLOCK_BODY(); \
-                                          THREAD_FENCE_MEM_ACQUIRE; /* spin waiting for my turn */ }
+                                       while (val != atomic_load_explicit((_Atomic uint32_t *)&(x)->s.ticket, memory_order_acquire)) SPINLOCK_BODY();\
+                                       THREAD_FENCE_MEM_ACQUIRE;}
```

Comment:
Calling `SPINLOCK_BODY` seems superfluous in this case.

Reply:
Maybe? If `QTHREAD_OVERSUBSCRIPTION` is enabled, `SPINLOCK_BODY` can yield back to the threading runtime we're using. On the other hand I'm not sure if `QTHREAD_OVERSUBSCRIPTION` is even usable anymore so that may be a moot point.

Looking at this again though, I think the `THREAD_FENCE_MEM_ACQUIRE` can be removed here. Or, more precisely either that is needed or the `memory_order_acquire` in the loop condition. Not both.

Reply:
I think we could remove `THREAD_FENCE_MEM_ACQUIRE`